src/core/test/index.js

```diff
@@ -20,9 +20,13 @@ export const runAsync = (testPromise) => {
         }
     };
 
-    const fail = () => {
+    const fail = (rejectionMessage) => {
+        const message = typeof rejectionMessage === 'string'
```

Comment:
IMO, statement should be the default - meaning that if the statement exists, we should always take it and not the rejection message. Not adding it is your way to communicate to passable that it should take the rejection message. Otherwise, why put it there in the first place?

Or maybe, what you think about is that your statement is a sort of a `catch all` message that will be used in case we can't get a rejection message?

Reply:
> Or maybe, what you think about is that your statement is a sort of a catch all message that will be used in case we can't get a rejection message?

Exactly. I think it would be better if the rejectionMessage would always take precedence and the statement would serve as a `fallback` (for example in a case of an uncaught error).

Reply:
I can relate to that logic. I can see some cases in which it would be unintuitive (but not counterintuitive), but these are edge cases and will happen due to human error.